# AT-ENCODE-pipeline
## What is AT-ENCODE:
<br>  Here we combine the published Arabidopsis epigenomic datasets (ChIP-seq, ATAC-seq, MNase-seq, BS-seq, RNA-seq and ncRNA-seq) and Hi-C data to construct a comprehensive Arabidopsis thaliana Encyclopedia of DNA Elements Database (AT-ENCODE). This database contains ~1800 datasets. The Arabidopsis TAIR10 is uniformly selected as the reference genome. So, it's convenient for the display and comparison of data. AT-ENCODE database mainly includes seven search functions, including "Histone Modification Search", "Transcriptome Search","Open Chromatin Region Search", "DNA methylation Search", "3D Genome", "Chromatin State" and "Wildtype vs Mutant Search", and they help database users to quickly search for targeted epigenetics. This also shows the epigenetic landscape of Arabidopsis in AT-ENCODE database from five aspects: histone modification, transcriptional expression, open chromatin state, DNA methylation degree and mutant difference. AT-ENCODE database has also equipped with WashU Epigenome Browser, it can show the standardized rice and Arabidopsis datasets more intuitively.</br>
### The AT-ENCODE pipeline have following dependencies :
* Python (2.7)
* samtools (>1.9)
* ChIA-PET tools (v3.0)
* BWA(0.7.17)
* MACS2(2.1.1)
* deepTools(2.5.3)
* fastp(0.20.0)
* cutadapt(1.9.1)
* bowtie(1.1.1)
* TopHat(v2.1.0)
* Cufflinks(2.2.1)
* HiC-Pro(2.11.4)
* BatMeth2(2.01)
<br></br>
### AT-ENCODE website:
[AT-ENCODE http://glab.hzau.edu.cn/AT-ENCODE/](http://glab.hzau.edu.cn/AT-ENCODE/)
